primitives/core/src/lib.rs

```diff
@@ -142,7 +142,10 @@ pub enum ServiceQuality {
 /// A trait which is called when the validation data is set.
 #[impl_trait_for_tuples::impl_for_tuples(30)]
 pub trait OnValidationData {
```

Comment:
We should change the name. I don't have any good idea, but it is clearly not only about `OnValidationData` anymore.

Reply:
I don't have ideas either... It is still called inside the `set_validation_data` function... so for me it still makes sense to keep the name, or at least is not so bad.

Reply:
```
/// Something that should be informed about system related events.
///
/// This includes events like [`on_validation_data`](Self::on_validation_data) that is being called when
/// the parachain inherent is executed that contains the validation data. Or like [`on_validation_code_applied`](Self::on_validation_code_applied) that is called when the new validation is written to the state. This means that
/// from the next block the runtime is being using this new code.
#[impl_trait_for_tuples::impl_for_tuples(30)]
trait OnSystemEvent {
	/// Called in each blocks once when the validation data is set by the inherent.
	fn on_validation_data(data: &PersistedValidationData) {}
	/// Called when the validation code is being applied, aka from the next block on this is the new runtime.
	fn on_validation_code_applied()  {}
}
```

Please move the trait to `parachain-system`. Please also rename the associated type to `OnSystemEvent`